service/controller/resource/values/create.go

```diff
@@ -55,20 +64,22 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 	}
 	r.logger.Debugf(ctx, "ensured secret %s/%s", secret.Namespace, secret.Name)
 
-	r.logger.Debugf(ctx, "updating App CR %#q with configmap and secret details", app.Name)
-	app.Spec.Config.ConfigMap = v1alpha1.AppSpecConfigConfigMap{
-		Namespace: configmap.Namespace,
-		Name:      configmap.Name,
-	}
-	app.Spec.Config.Secret = v1alpha1.AppSpecConfigSecret{
-		Namespace: secret.Namespace,
-		Name:      secret.Name,
-	}
-	err = r.k8sClient.CtrlClient().Update(ctx, &app)
-	if err != nil {
-		return microerror.Mask(err)
+	if app.Spec.Config.ConfigMap.Name == "" || app.Spec.Config.Secret.Name == "" {
```

Comment:
This `if` should be stronger. We should check if the values set are what we want, not only if they are empty.